CircleCI config - Remove iOS jobs

geriux opened the pull request:
Removes CircleCI iOS jobs in favor of these being migrated to Buildkite.

To test the rest of the CI checks should pass.

**NOTE:** The iOS-required CircleCI check was removed and the Builkite check `Test iOS on Device - Canaries` was added.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/RELEASE-NOTES.txt) if necessary.

Commit:
CircleCI config - Remove iOS jobs

twstokes reviewed (approved):
These changes make sense to me @geriux! 🚀 

> and the Builkite check Test iOS on Device - Canaries was added

This only thing I'm not sure about is this part. How were they added?

Thanks!

twstokes commented:
Also want to cc @mokagio just to stay looped in. 👍

geriux commented:
> This only thing I'm not sure about is this part. How were they added?

Sorry I should've expanded that 😅 , I've updated the `trunk` branch settings on GitHub. The old CircleCI required check for iOS has been removed for merging pull requests, and I've added the one from Buildkite.

mokagio reviewed (commented):
Thanks @geriux 🙇‍♂️
